### DISCUSSION TITLE
STM32U5

### DISCUSSION THREAD

**Discussion by Maerdl** [hathach/tinyusb#2079] [Q&A]
Hi,

I have a STM32U585 MB1551C board running with a custom software.
I have a tinyusb stack up and running. The device successfully appears as a vcom port on a windows pc.
But unfortunately after some time, round about 10-20sec., I get another USB_RST interrupt which causes a TU_ASSERT in state DCD_EVENT_XFER_COMPLETE in file usbd.c

This assert happens because _usbd_dev will be zeroed by function configuration_reset.

Following log is printed, just added the last few lines:

USBD Xfer Complete on EP 83 with 0 bytes
  MSC xfer callback
  SCSI Data [Lun0]
USBD Bus Reset : Full Speed
USBD Xfer Complete on EP 83 with 0 bytes
tud_task_ext 553: ASSERT FAILED
 
I have some questions on which i'm totally lost:
* why is there a USB_RST after some time ?
* what else can i do wrong ?

i'm sitting on the master branch with #5add4c97

Kind regards
Martin

**Comment by hathach** [comment]
bus reset is a physical line condition, probably driven by host. You should check PC kernel message to try figuring out why host do a reset. If it is not from host, then it is from software, does the issue occur when running with one of the stock example ?

**Comment by Maerdl** [comment]
Thanks
it was a bad configuration in the configuration
